src/libraries/System.Private.CoreLib/src/System/String.Manipulation.cs

```diff
@@ -1672,11 +1671,10 @@ private unsafe string InternalSubString(int startIndex, int length)
 
             string result = FastAllocateString(length);
 
-            fixed (char* dest = &result._firstChar)
-            fixed (char* src = &_firstChar)
-            {
-                wstrcpy(dest, src + startIndex, length);
-            }
+            Buffer.Memmove(
```

Comment:
Does this function still need to be marked `unsafe`?

Reply:
No. Good catch!

Reply:
`unsafe` is a pure C# language construct today. It does not control whether it is ok to call unsafe APIs. It means that `unsafe` tells nothing about whether the code is unsafe. See  https://github.com/dotnet/runtime/issues/31354 for more details.